### ISSUE TITLE
Questions about implementing hover and timer functionalities

### ISSUE THREAD

**Issue by math-artist** [pyvista/pyvistaqt#65]
Hi,

I am trying to implement a simple hover feature that could be used to execute a callback after the mouse is not moving and hover an actor for a certain period of time. My end goal: ray tracing on hover to display a scalar from a dataset on the fly. This could be flexible and work like the vtkBalloonWidget, or this could be a fixed 2D actor in the corner or just getting the result for somewhere else in Qt.

I am currently using pyvista with Qt, but I would prefer to make a solution without Qt and put that in a PR. I based my first tests on this example:
[https://lorensen.github.io/VTKExamples/site/Cxx/Widgets/HoverWidget/](https://lorensen.github.io/VTKExamples/site/Cxx/Widgets/HoverWidget/)
But I am getting a very strange effect: every action with the mouse is perpetual unless I release the button. For example, panning is happening without moving the mouse.

Anybody knows if it is possible to do it without Qt if I am using VTK in a Qt widget?
Any idea of what is happening? ~~This simple code set with iren is not printing (no callback) but it seems to overwrite something.~~
```
def hover_widget(render_window_interactor):
    widget = vtk.vtkHoverWidget()
    widget.SetTimerDuration(300)
    widget.SetInteractor(render_window_interactor)
    widget.AddObserver(vtk.vtkCommand.TimerEvent, callback)
    widget.AddObserver(vtk.vtkCommand.EndInteractionEvent, callback)
    widget.On()
    return widget


def callback(object, event_id):
    print("hover now")
```

Edit: okay, I added a return line to save the widget (facepalm here) and I tested with pyvista only. It works! but only without Qt.  In my app the callback is firing all the time. I will investigate further what is happening in pyvistaqt.

**Comment by math-artist** [comment]
thanks @banesullivan. I didn't find the option to transfer.

I had to abandon the idea of using vtkHoverWidget for now, just because I have no idea what is happenning. My fear is that using QtInteractor makes some VTK functions unusable. 

The good news is that it was simple to implement with Qt.

Here's a preview of the result:
[https://www.loom.com/share/5039ec0f28fb49f0a96360137eeeba6c?from_recorder=1](https://www.loom.com/share/5039ec0f28fb49f0a96360137eeeba6c?from_recorder=1)

I made an override for mouseMouveEvent to start a timer if a callback is set. It displays a cell form a structured grid linked to a mesh and display information for that cell in a vtkTextWidget. Cursor needs to stop to trigger the event, so it is quite light, but it feels a bit laggy from a user perspective.

```
    def set_mouse_move_callback(self, callback, delay=300):
        self.mouse_timer = QtCore.QTimer()
        self.mouse_timer.setInterval(delay)
        self.mouse_timer.setSingleShot(True)
        self.mouse_timer.timeout.connect(callback)

    def mouseMoveEvent(self, ev):
        if self.mouse_timer:
            self.mouse_timer.start()
        super().mouseMoveEvent(ev)
```

Is there anything there that would be worth adding to pyvista or pyvistaqt? I have made a bunch of implementations to annotate  points or display 2D text. They are usually not that hard to play with initially, but they are a pain to format properly, with missing controls for padding, etc...
